serverlessworkflow/modules/ROOT/pages/eventing/event-correlation-with-workflows.adoc

```diff
@@ -1 +1,216 @@
-//= Event correlation with workflows
\ No newline at end of file
+= Event correlation with workflows
+=====================================
+:compat-mode!:
+// Metadata:
+:description: Event Correlation in Serverless Workflow
+:keywords: kogito, workflow, serverless, correlation, association, domain, events
+
+[[ref-example-jq-expression-switch-conditions]]
+
+Event correlation plays a significant role in large event-driven applications. It allows matching one or more events with a particular workflow instance. Correlation rules with domain information can be used when defining consumed events to match a workflow instance. It is a more practical alternative to the internal identifier `processInstanceId` since you can use any external domain identifier as a correlation.
+
+A correlation definition consists of one or more attributes associated with an event and the respective workflow to which it belongs.  As described in the link:{spec_doc_url}#correlation-definition[Serverless Workflow Specification], the correlation property defines the possible correlations for a given event. Each element should contain a `contextAttributeName` set for the value that should match an attribute from an event.
+
+You can optionally set a `contextAttributeValue` property that should match the value for the respective attribute defined in `contextAttributeName` for the consumed events in a workflow.
+
+[NOTE]
+====
+The incoming events consumed by the workflow engine should contain the correlation attributes, set in the definition, as extension context attributes, in compliance with the link:{cloud_events_url}[Cloud Event] format, thus they are not part of the event payload.
+====
+
+When starting a new workflow instance, the correlation attributes set in the event definition are extracted from the event and associated with the workflow instance which is being created. Thus, the start event does not trigger a correlation evaluation, instead, it is where correlation attributes and values are set, which are used to evaluate against other incoming events that might trigger this instance.
+
+Once an event is consumed and the correlation rules are evaluated, if any workflow instance matches them, the engine continues the execution of that matched instance.
```

Comment:
As I said previously, I would maybe add what happens before the actual start - that an event with the correlation attributes must be sent from somewhere and is received by the workflow service.

Reply:
Ok, I inserted one more sentence to clarify this part and a link to the Event State guide.
Thanks @MarianMacik